gpMgmt/bin/lib/pexpect/pxssh.py

```diff
@@ -265,21 +265,21 @@ def login (self, server, username, password='', terminal_type='ansi',
         # This does not distinguish between a remote server 'password' prompt
         # and a local ssh 'passphrase' prompt (for unlocking a private key).
         spawn._spawn(self, cmd)
-        i = self.expect(["(?i)are you sure you want to continue connecting", original_prompt, "(?i)(?:password)|(?:passphrase for key)", "(?i)permission denied", "(?i)terminal type", TIMEOUT, "(?i)connection closed by remote host"], timeout=login_timeout)
+        i = self.expect(["(?i)are you sure you want to continue connecting", original_prompt, "(?i)(?:password)|(?:passphrase for key)", "(?i)permission denied", "(?i)terminal type\?", TIMEOUT, "(?i)connection closed by remote host"], timeout=login_timeout)
```

Comment:
Did everything work before without a question mark at the end?

Reply:
`Expect` call always reacts on the first match it finds, and since input arrives in unpredictable chunks, there is always a possibility that similar string will apear before real prompt. Something like that already happend in pexpect upstream with [password case](https://github.com/pexpect/pexpect/pull/452).  And answering your question, if there isn't unexpected input from ssh, then everything works well. This patch tries to cover unexpected cases just partially, hoping that `?` sign will only appear once.